yt/funcs.py

```diff
@@ -99,13 +115,7 @@ def read_struct(f, fmt):
 
 def just_one(obj):
     # If we have an iterable, sometimes we only want one item
-    if hasattr(obj, "flat"):
-        if isinstance(obj, YTArray):
-            return YTQuantity(obj.flat[0], obj.units, registry=obj.units.registry)
-        return obj.flat[0]
-    elif iterable(obj):
-        return obj[0]
-    return obj
+    return first(collapse(obj))
```

Comment:
Do we expect collapse() to handle the YTQuantity and unit registry passing as this function previously did?

Reply:
Just added a unit test to address this, but I'm honestly not 100% sure I'm covering the cases that actually matter.

Reply:
I think the test looks good. 🙂